### ISSUE TITLE
gifsave: animation with alpha channel image artifacts on resize

### ISSUE THREAD

**Issue by dloebl** [libvips/libvips#3090]
We noticed image artifacts when creating thumbnails of some GIFs with an alpha channel.
Interestingly, the issue seems to occur only when libvips is configured with [libimagequant](https://github.com/ImageOptim/libimagequant) - with [quantizr](https://github.com/DarthSim/quantizr) the artifacts are not present.

**Reproducer:**
[input.gif](https://user-images.githubusercontent.com/34104349/194905246-1eb92dde-cdf4-49fa-9060-1446f996971c.gif)
```
vipsthumbnail --size 280x input.gif[n=-1] -o out.gif
```

| original | thumbnail |
| - | - |
| ![input.gif](https://user-images.githubusercontent.com/34104349/194905246-1eb92dde-cdf4-49fa-9060-1446f996971c.gif) |  ![out](https://user-images.githubusercontent.com/34104349/194905317-f4335276-de97-4b93-8e70-1595d080b0db.gif) |

**libvips version:** master (5fcca56fe832359eb75fe6cc5d52ab79522726fe)
**libimagequant version:** v4.0.4

**Comment by jcupitt** [comment]
That's a strange one. @DarthSim do you have any thoughts on this?

**Comment by lovell** [comment]
@dloebl Are you able to check if this only affects libimagequant v4+ (I couldn't reproduce locally with the latest v2)?

**Comment by dloebl** [comment]
@lovell Yes, indeed: Looks like only libimagequant >= v4 is affected:
| libimagequant v4.0.0 | libimagequant v2.17.0 |
| - | - |
| ![4](https://user-images.githubusercontent.com/34104349/196396900-01f3c8ff-f8e1-44f8-8a51-9e341e5e2ab6.gif) | ![2](https://user-images.githubusercontent.com/34104349/196396935-91ddc65a-8289-423f-8475-e804ac913ae5.gif) |

**Comment by jcupitt** [comment]
Then I guess we can ignore this, is that right?

**Comment by dloebl** [comment]
OK let's close this issue for now. I'll open a new issue in case I find something related to libvips.

**Comment by DarthSim** [comment]
Yup, it's definitely a bug of LIQ 4. I added a debug output of a generated palette, and here's what I see:

```
0: 0 0 0 0
1: 0 0 0 0
2: 255 255 255 255
3: 0 0 0 255
4: 19 19 19 255
...
```

The first two colors are the same for some reason. It seems like LIQ picks the second color for the area that becomes black, but GIF can use only one color for transparency.

**Comment by dloebl** [comment]
I reported this issue on libimagequant (https://github.com/ImageOptim/libimagequant/issues/101) and it has been resolved with v4.0.5.
